Pico/Motors.cpp

```diff
@@ -43,117 +40,137 @@ void Motors::GetAllSpeeds()
     Serial.println(motor4.GetSpeed());
 };
 
-void Motors::StopMotors()
-{
+void Motors::StopMotors() {
     motor1.StopMotor();
     motor2.StopMotor();
     motor3.StopMotor();
     motor4.StopMotor();
 };
 
-void Motors::MoveForward()
-{
+void Motors::MoveForward() {
     motor1.MoveForward();
-    motor1.StopMotor();
     motor2.MoveForward();
-    motor2.StopMotor();
     motor3.MoveForward();
-    motor3.StopMotor();
     motor4.MoveForward();
-    motor4.StopMotor();
 };
 
-void Motors::MoveRight()
-{
+void Motors::MoveRight() {
     StopMotors();
     motor1.MoveForward();
     motor2.MoveBackward();
     motor3.MoveBackward();
-    motor4.MoveForward();
+    motor4.MoveBackward();
 };
 
-void Motors::MoveLeft()
-{
+void Motors::MoveLeft() {
     StopMotors();
     motor1.MoveBackward();
     motor2.MoveForward();
     motor3.MoveForward();
-    motor4.MoveBackward();
+    motor4.MoveForward();
 };
 
-void Motors::MoveBackward()
-{
+void Motors::MoveBackward() {
     StopMotors();
     motor2.MoveBackward();
     motor3.MoveForward();
 };
 
-void Motors::MoveMotor1()
-{
+void Motors::MoveMotor1() {
     motor1.MoveForward();
 };
 
-void Motors::MoveMotor2()
-{
+void Motors::MoveMotor2() {
     motor2.MoveForward();
 };
 
-void Motors::MoveMotor3()
-{
+void Motors::MoveMotor3() {
     motor3.MoveForward();
 };
 
-void Motors::MoveMotor4()
-{
+void Motors::MoveMotor4() {
     motor4.MoveForward();
 };
 
-void Motors::MoveMotors(int degree, uint8_t speed)
-{
-    float m1 = cos(((45 + degree) * PI / 180));
-    float m2 = cos(((135 + degree) * PI / 180));
-    float m3 = cos(((225 + degree) * PI / 180));
-    float m4 = cos(((315 + degree) * PI / 180));
-    int speedA = abs(int(m1 * speed));
-    int speedB = abs(int(m2 * speed));
-    int speedC = abs(int(m3 * speed));
-    int speedD = abs(int(m4 * speed));
+void Motors::MoveMotors(int degree, uint8_t speed) {
+    float m1 = cos(((45+degree) * PI / 180));
+    float m2 = cos(((135+degree) * PI / 180));
+    float m3 = cos(((225+degree) * PI / 180));
+    float m4 = cos(((315+degree) * PI / 180));
+    int speedA = abs(int(m1*speed));
+    int speedB = abs(int(m2*speed));
+    int speedC = abs(int(m3*speed));
+    int speedD = abs(int(m4*speed));
+
+    analogWrite(motor1.GetSpeed(), speedA);
```

Comment:
You should abstract the behavior of setting each Motor PWM and direction inside a function of the Motor class, instead of setting the speed in this method.

A requested the same for soccer lightweight, for further guidance, take a look at this commit where they implemented it: https://github.com/RoBorregos/robocup-soccer-light-2024/pull/15/commits/6bcfdf9f47fe70442b94f6178fd299a194447c72